src/Microsoft.Framework.Runtime/DependencyManagement/LockFile.cs

```diff
@@ -15,13 +16,34 @@ public class LockFile
         public IList<LockFileLibrary> Libraries { get; set; } = new List<LockFileLibrary>();
         public IList<LockFileTarget> Targets { get; set; } = new List<LockFileTarget>();
 
+        public IEnumerable<ICompilationMessage> GetDiagnostics(Project project)
+        {
+            string message;
+            if (!IsValidForProject(project, out message))
+            {
+                yield return new FileFormatMessage(
+                    $"{message}. Please run \"dnu restore\" to generate a new lock file.",
+                    Path.Combine(project.ProjectDirectory, LockFileReader.LockFileName),
+                    CompilationMessageSeverity.Error);
+            }
+        }
+
         public bool IsValidForProject(Project project)
         {
+            string message;
+            return IsValidForProject(project, out message);
```

Comment:
We'll generally want the string, at least in the hot path (runtime startup). Plus, the big string isn't actually allocated until an error occurs, this is just a pointer to hold it :). So I'm not too worried about this piece of code.